extensions/notion/src/components/forms/CreatePageForm.tsx

```diff
@@ -0,0 +1,234 @@
+import { ActionPanel, Clipboard, Icon, Form, showToast, useNavigation, Action, Toast } from "@raycast/api";
+import { useForm } from "@raycast/utils";
+import { useState, useEffect } from "react";
+
+import {
+  useDatabaseProperties,
+  useDatabases,
+  useDatabasesView,
+  useRecentPages,
+  useRelations,
+  useUsers,
+} from "../../hooks";
+import { createDatabasePage, DatabaseProperty } from "../../utils/notion";
+import { handleOnOpenPage } from "../../utils/openPage";
+import { ActionSetVisibleProperties } from "../actions";
+
+import { createConvertToFieldFunc, FieldProps } from "./PagePropertyField";
+
+export type CreatePageFormValues = {
+  database: string | undefined;
+  [K: string]: Form.Value | undefined;
+};
+
+type CreatePageFormProps = {
+  mutate?: () => Promise<void>;
+  defaults?: CreatePageFormValues;
+};
+
+export function CreatePageForm({ mutate, defaults }: CreatePageFormProps) {
+  const initialDatabaseId = defaults?.database;
+
+  const [databaseId, setDatabaseId] = useState<string | null>(initialDatabaseId ? initialDatabaseId : null);
+  const { data: databaseView, setDatabaseView } = useDatabasesView(databaseId || "__no_id__");
+  const { data: databaseProperties } = useDatabaseProperties(databaseId);
+  const { data: users } = useUsers();
+  const { data: databases, isLoading: isLoadingDatabases } = useDatabases();
+  const { data: relationPages, isLoading: isLoadingRelationPages } = useRelations(databaseProperties);
+
+  const initialValues: Partial<CreatePageFormValues> = { database: databaseId ?? undefined };
+  for (const { id, type } of databaseProperties) {
+    let value = defaults?.[id];
+    if (type == "date" && value) value = new Date(value as string);
+    const key = "property::" + type + "::" + id;
+    initialValues[key] = value;
+  }
+
+  const { itemProps, values, handleSubmit } = useForm<CreatePageFormValues>({
+    initialValues,
+    async onSubmit(values) {
+      const titleKey = Object.keys(values).find((key) => key.includes("property::title"));
+      if (!titleKey || !values[titleKey]) {
+        showToast({
+          style: Toast.Style.Failure,
+          title: "Title Required",
+          message: "Please set title value",
+        });
+        return;
+      }
```

Comment:
Note that you could use the `validation` object for that: https://developers.raycast.com/utilities/react-hooks/useform#formvalidation

Reply:
Good idea